core/txpool/celo_validation.go

```diff
@@ -0,0 +1,133 @@
+package txpool
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var NonWhitelistedFeeCurrencyError = errors.New("Fee currency given is not whitelisted at current block")
+
+// FeeCurrencyValidator validates currency whitelisted status at the specified
+// block number.
+type FeeCurrencyValidator interface {
+	IsWhitelisted(st *state.StateDB, feeCurrency *common.Address) bool
+	// Balance returns the feeCurrency balance of the address specified, in the given state.
+	// If feeCurrency is nil, the native currency balance has to be returned.
+	Balance(st *state.StateDB, address common.Address, feeCurrency *common.Address) *big.Int
+}
+
+func NewFeeCurrencyValidator() FeeCurrencyValidator {
+	return &feeval{}
+}
+
+type feeval struct {
+}
```

Comment:
I assume this is going to have state in a later commit (a cache of the whitelist?). Otherwise, `IsWhitelisted` and `Balance` should just be free standing functions.

Reply:
Yes, I'm leaving this as a placeholder for whatever we need to call the contracts for checking whitelisted currencies and balances by state.